src/symbol_object.cpp

```diff
@@ -59,13 +59,16 @@ Value SymbolObject::to_proc_block_fn(Env *env, Value self_value, size_t argc, Va
 Value SymbolObject::cmp(Env *env, Value other_value) {
     if (!other_value->is_symbol()) return NilObject::the();
     SymbolObject *other = other_value->as_symbol();
+    if (this == other)
+        return Value::integer(0);
     int diff = strcmp(m_name, other->m_name);
     int result;
     if (diff < 0) {
         result = -1;
     } else if (diff > 0) {
         result = 1;
     } else {
+        // NOTE: This should not happen
```

Comment:
You're right that this shouldn't happen. Let's throw a `NAT_UNREACHABLE()` in here to make sure it never does.